RCTAppleHealthKit/RCTAppleHealthKit+Methods_Workout.m

```diff
@@ -12,6 +12,48 @@
 
 @implementation RCTAppleHealthKit (Methods_Workout)
 
+- (void)workout_getAnchoredQuery:(NSDictionary *)input callback:(RCTResponseSenderBlock)callback
+{
+    HKUnit *unit = [RCTAppleHealthKit hkUnitFromOptions:input key:@"unit" withDefault:[HKUnit countUnit]];
+    NSUInteger limit = [RCTAppleHealthKit uintFromOptions:input key:@"limit" withDefault:HKObjectQueryNoLimit];
+    
+    HKQueryAnchor *anchor = [RCTAppleHealthKit hkAnchorFromOptions:input];
+    NSString *type = [RCTAppleHealthKit stringFromOptions:input key:@"type" withDefault:@"Walking"];
+    NSDate *startDate = [RCTAppleHealthKit dateFromOptions:input key:@"startDate" withDefault:[NSDate date]];
+    NSDate *endDate = [RCTAppleHealthKit dateFromOptions:input key:@"endDate" withDefault:[NSDate date]];
+    
+    NSPredicate *predicate = [HKQuery predicateForSamplesWithStartDate:startDate endDate:endDate options:HKQueryOptionStrictStartDate];
```

Comment:
Do we still need the start date and end date if we are using the anchor? Or it is so that we are backward compatible?

Reply:
Yup exactly. Also, in case the developer wants to use the query without the anchor, they can still use the query using a given time frame.

Reply:
So the issue that we ran into was if the sync happens, and then a new workout is added with start time before the last sync time it won't be synched. I think we should be moving away from time based.

Reply:
The anchor query will return the new workout even the query date stays the same.